code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -744,7 +745,19 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 			to_chat(user, "You apply the enhancer. It now has triple the amount of uses.")
 			Uses = 3
 			enhanced = 1
-			qdel (O)
+			qdel(O)
+			O = null
+
+		//slime res
+		if(istype(O, /obj/item/weapon/slimeres))
+			if(Uses == 0)
+				to_chat(user, "<span class='warning'>The solution doesn't work on used extracts!</span>")
+				return ..()
+			to_chat(user, "You splash the Slime Resurrection Serum onto the extract causing it to quiver and come to life.")
+			var/mob/living/carbon/slime/S = new primarytype
```

Comment:
You can set the `loc` directly by passing it as an argument to new:

``` DM
var/mob/living/carbon/slime/S = new primarytype(get_turf(src))
```

Could you go and do that for all the other news?